autorest/models/schema.py

```diff
@@ -24,14 +24,93 @@
 #
 # --------------------------------------------------------------------------
 
-class CompositeType:
-    """Represent a composite type schema ready to be serialized in Python.
+
+class ComposedProperty:
+    """Represents a property of a CompositeType
     """
+    def __init__(self, name, description, property_type, **kwargs):
+        self.name = name
+        self.description = description
+        self.property_type = property_type
+        self.required = kwargs.pop('required', False)
+        self.readonly = kwargs.pop('readonly', False)
+        self.constant = kwargs.pop('constant', False)
+        self.default_value = kwargs.pop('default_value', None)
+
+    def get_property_documentation_string(self):
+        if not self:
+            raise TypeError("Property can not be none.")
```

Comment:
Surprising test?

Reply:
Yeah, I see this is redundant, I'm going to get rid of it. Thanks!